apps/angular/59-content-projection-defer/src/app/page-2.ts

```diff
@@ -34,10 +37,20 @@ interface Post {
   `,
   changeDetection: ChangeDetectionStrategy.OnPush,
   imports: [ExpandableCard],
+  standalone: true,
 })
 export class Page2 {
-  public postRessource = httpResource<Post[]>(
-    'https://jsonplaceholder.typicode.com/posts',
-  );
+  private readonly injector = inject(Injector);
+  public postResource: ReturnType<typeof httpResource<Post[]>> | null = null;
   protected readonly ResourceStatus = ResourceStatus;
+
+  loadPosts() {
+    if (!this.postResource) {
+      runInInjectionContext(this.injector, () => {
```

Comment:
using `runInInjectionContext` is often the sign of a bad desing. You should avoid using it at all cost